### PR TITLE
Update Calico Enterprise Compliance to use Linseed

### PR DESCRIPTION
**Pull request by caseydavenport**
## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/compliance/compliance_controller.go [hunk 1]
```diff
@@ -364,11 +364,41 @@ func (r *ReconcileCompliance) Reconcile(ctx context.Context, request reconcile.R
 		r.status.SetDegraded(operatorv1.ResourceNotReady, "Elasticsearch gateway certificate are not available yet, waiting until they become available", nil, reqLogger)
 		return reconcile.Result{}, nil
 	}
-	trustedBundle := certificateManager.CreateTrustedBundle(managerInternalTLSSecret, esgwCertificate)
+	linseedCertificate, err := certificateManager.GetCertificate(r.client, render.TigeraLinseedSecret, common.OperatorNamespace())
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceValidationError, fmt.Sprintf("Failed to retrieve / validate  %s", render.TigeraLinseedSecret), err, reqLogger)
+		return reconcile.Result{}, err
+	} else if esgwCertificate == nil {
```
Comment (asincu) [inline pkg/controller/compliance/compliance_controller.go:371]: Should we check if the linseed certificate is nil ?

File: pkg/render/compliance.go [hunk 2]
```diff
@@ -374,11 +388,19 @@ func (c *complianceComponent) complianceControllerClusterAdminClusterRoleBinding
...
+		// This should never be nil, but we check it anyway just to be safe.
+		keyPath, certPath = c.cfg.ControllerKeyPair.VolumeMountKeyFilePath(), c.cfg.ControllerKeyPair.VolumeMountCertificateFilePath()
+	}
+
 	envVars := []corev1.EnvVar{
 		{Name: "LOG_LEVEL", Value: "info"},
 		{Name: "TIGERA_COMPLIANCE_JOB_NAMESPACE", Value: ComplianceNamespace},
 		{Name: "TIGERA_COMPLIANCE_MAX_FAILED_JOBS_HISTORY", Value: "3"},
 		{Name: "TIGERA_COMPLIANCE_MAX_JOB_RETRIES", Value: "6"},
+		{Name: "LINSEED_CLIENT_CERT", Value: certPath},
```
Comment (asincu) [inline pkg/render/compliance.go:402]: Don;t we also need the CA passed in ?

File: pkg/render/compliance.go [hunk 3]
```diff
@@ -488,10 +511,18 @@ func (c *complianceComponent) complianceReporterClusterRoleBinding() *rbacv1.Clu
...
+	if c.cfg.ReporterKeyPair != nil {
+		// This should never be nil, but we check it anyway just to be safe.
+		keyPath, certPath = c.cfg.ReporterKeyPair.VolumeMountKeyFilePath(), c.cfg.ReporterKeyPair.VolumeMountCertificateFilePath()
+	}
+
 	dirOrCreate := corev1.HostPathDirectoryOrCreate
 	envVars := []corev1.EnvVar{
 		{Name: "LOG_LEVEL", Value: "info"},
 		{Name: "TIGERA_COMPLIANCE_JOB_NAMESPACE", Value: ComplianceNamespace},
+		{Name: "LINSEED_CLIENT_CERT", Value: certPath},
```
Comment (asincu) [inline pkg/render/compliance.go:524]: Same as above: don;t we need the CA ?

File: pkg/controller/compliance/compliance_controller.go [hunk 4]
```diff
@@ -364,11 +364,41 @@ func (r *ReconcileCompliance) Reconcile(ctx context.Context, request reconcile.R
 		r.status.SetDegraded(operatorv1.ResourceNotReady, "Elasticsearch gateway certificate are not available yet, waiting until they become available", nil, reqLogger)
 		return reconcile.Result{}, nil
 	}
-	trustedBundle := certificateManager.CreateTrustedBundle(managerInternalTLSSecret, esgwCertificate)
+	linseedCertificate, err := certificateManager.GetCertificate(r.client, render.TigeraLinseedSecret, common.OperatorNamespace())
```
Comment (asincu) [inline pkg/controller/compliance/compliance_controller.go:367]: Don't we also need to add this certificate to the trusted bundled built for Linseed ?

File: pkg/render/compliance.go (hunk 2 above)
Comment (caseydavenport) [reply pkg/render/compliance.go:402]: The CA is defaulted in the code, so we don't need to set that explicitly here.

Comment (caseydavenport) [reply pkg/render/compliance.go:402]: That's how it has been done for other components as well

File: pkg/controller/compliance/compliance_controller.go (hunk 1 above)
Comment (caseydavenport) [reply pkg/controller/compliance/compliance_controller.go:371]: good catch

**Review by asincu — APPROVED** [review]

**Pull request merged**